services/multitenant/tenantstats_test.go

```diff
@@ -44,117 +49,280 @@ var _ = Describe("tenantStats", func() {
+				input[workspaceID3][destType1] = addJobWID3
+				tenantStats.ReportProcLoopAddStats(input, "rt")
+				tenantStats.lastDrainedTimestamps[workspaceID1][destType1] = time.Now().Add(time.Duration(-1) * time.Minute)
+				for i := 0; i < int(metric.AVG_METRIC_AGE); i++ {
+					tenantStats.UpdateWorkspaceLatencyMap(destType1, workspaceID1, 0.1)
+					tenantStats.UpdateWorkspaceLatencyMap(destType1, workspaceID2, 0.11)
+					tenantStats.UpdateWorkspaceLatencyMap(destType1, workspaceID3, 0.12)
+				}
+				routerPickUpJobs, usedLatencies := tenantStats.GetRouterPickupJobs(destType1, noOfWorkers, routerTimeOut, 10000, timeGained)
+				// Jobs are picked from Input Loop
+				// workspaceID1 has the least latency but also the least priority
+				// as it has draining jobs withing the last minute
+				// The algorithm tries to pick the jobs likely to succeed
+				// So the other two workspaces had the maximum picked up jobs(1000)
+				Expect(routerPickUpJobs[workspaceID1]).To(Equal(6019))
+				Expect(routerPickUpJobs[workspaceID2]).To(Equal(1000))
+				Expect(routerPickUpJobs[workspaceID3]).To(Equal(1000))
+				Expect(almostEqual(usedLatencies[workspaceID1], 0.1)).To(BeTrue())
+				Expect(almostEqual(usedLatencies[workspaceID2], 0.11)).To(BeTrue())
+				Expect(almostEqual(usedLatencies[workspaceID3], 0.12)).To(BeTrue())
+			})
+
+			It("Customer with high failure rate should be de-prioritised", func() {
+				addJobWID1 := 1000
+				addJobWID2 := 1000
+				addJobWID3 := 1000
+				input := make(map[string]map[string]int)
+				input[workspaceID1] = make(map[string]int)
+				input[workspaceID2] = make(map[string]int)
+				input[workspaceID3] = make(map[string]int)
+				input[workspaceID1][destType1] = addJobWID1
+				input[workspaceID2][destType1] = addJobWID2
+				input[workspaceID3][destType1] = addJobWID3
+				//Explicitly Making the Input Rate Lower to pick from PileUp
+				tenantStats.processorStageTime = time.Now().Add(time.Duration(-1) * time.Minute)
+				tenantStats.ReportProcLoopAddStats(input, "rt")
+				for i := 0; i < int(metric.AVG_METRIC_AGE); i++ {
+					tenantStats.UpdateWorkspaceLatencyMap(destType1, workspaceID1, 0.4)
+					tenantStats.UpdateWorkspaceLatencyMap(destType1, workspaceID2, 0.41)
+					tenantStats.UpdateWorkspaceLatencyMap(destType1, workspaceID3, 0.42)
+					//Making the failure rate for workspaceID1 high
+					tenantStats.CalculateSuccessFailureCounts(workspaceID1, destType1, false, false)
+				}
+				routerPickUpJobs, usedLatencies := tenantStats.GetRouterPickupJobs(destType1, noOfWorkers, routerTimeOut, 10000, timeGained)
+				// Jobs are picked from Input Loop
+				// Router In Rate was very low so the jobs had to be picked up from PileUp
+				// workspaceID1 has the highest failure rate so the least priority in PileUp
+				// The algorithm tries to pick the jobs likely to succeed from PileUp
+				// So the other two workspaces had the maximum picked up jobs(1000)
+				fmt.Println(routerPickUpJobs)
```

Comment:
If output is needed we can use `t.Log()` which respect `testing.Verbose()`.

Reply:
I missed removing the print statements post debugging. Let me remove this